### DISCUSSION TITLE
Markdown Style & Linting Fixes?

### DISCUSSION THREAD

**Discussion by kraker** [livialima/linuxupskillchallenge#4] [Ideas]
@livialima Would you be interested in stylistic edits and linting of the markdown files in this repository?

For example, setting an 80 character width limit would help with readability of the raw markdown text and using [fenced code blocks](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks) instead of indented blocks would allow us to utilize syntax highlighting.

These are a couple of things that jumped out at me when working with the project. Running the code-base through a markdown linter might catch most of it.

If you're open to it, I would get started working on a PR for this.

**Comment by livialima** [comment]
Yes, please! Any help in the readability of the lessons is very welcomed.

**Reply by kraker** [reply]
Only 1 year later, but I got a wild hair this weekend and I'm working on setting up some scaffolding to provide linting support to the project. This is going to be somewhat of a large commit because in my opinion if we're doing this right we're managing linting as project dependencies and adding it to GH workflows so that it's part of the build automation. And then there's the linting of all of the various files to bring them into form.

Before I get too ahead of ourselves, I submit a draft of my PR for preview: https://github.com/livialima/linuxupskillchallenge/pull/16

The new `CONTRIBUTING.md` is probably a good entry-point to the scaffolding I'm setting up which relies heavily on the pre-commit tool.

I've linted a few things, but most of the project markdown still needs to be linted. Maybe that's another PR, if this is getting too big...

@livialima let me know what you think.

**Reply by kraker** [reply]
Actually, @livialima, PR is ready for review. I think it's large enough already and I think linting the rest of the repo might be several more PR's depending...

https://github.com/livialima/linuxupskillchallenge/pull/16

Also, I'm not sure if I'll be able to get to linting the markdown in docs/ right away.

**Reply by livialima** [reply]
I saw that coming, just didn't sit to look through it all yet. It looks good at a first quick glance, but since we're changing several files, I'm taking more time to review. Thanks for the patience and thank you so much for the contribution!
